openshift/jag-csrs.yml

```diff
@@ -0,0 +1,164 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: ${APP_NAME}
+  description: Deployment template for the JAG-csrs
+  
+labels:
+  app: ${APP_NAME}
+  app.kubernetes.io/name: ${COMPONENT_NAME}
+  app.kubernetes.io/part-of: ${APP_NAME}
+
+  
+parameters:
+
+- name: APP_NAME
+  description: The application name that each component is part of
+  required: true
+  value: jag-csrs-frontend
+
+- name: SPLUNK_CONFIG_NAME
+  description: The application name that each component is part of
+  required: true
+  value: blahblah
+
+- name: ROUTE_PREFIX
+  description: The application name that each component is part of
+  required: true
+  value: jag-csrs-frontend
+
+- name: COMPONENT_NAME
+  description: The component name
+  required: true
+  value: jag-csrs-frontend
+  
+- name: OC_NAMESPACE
+  description: The OpenShift namespace prefix
+  required: true
+  # Below value must be changed as per gold cluster oc nameplate
+  value: blahblah
+  
+- name: OC_ENV
+  description: The OpenShift environment, ie dev, test or prod
+  required: true
+  value: dev
+
+- name: DNS
+  description: The application name that each component is part of
+  required: false
+  # Below value must be changed as per gold cluster certificate/DNS
+  value: apps.silver.devops.gov.bc.ca
+
+- name: BCEID_URL
+  description: BCEID_URL for dev, test or prod
+  required: true
+  value: https://www.development.bceid.ca/
+  
+objects:
+
+### JAG-csrs Deployment Config ###
+- apiVersion: apps.openshift.io/v1
+  kind: DeploymentConfig
+  metadata:
+    name: ${COMPONENT_NAME}
+    namespace: ${OC_NAMESPACE}-${OC_ENV}
+      
+  spec:
+    completionDeadlineSeconds: 1200
+    replicas: 1
+    selector:
+      name: ${COMPONENT_NAME}
+    strategy:
+      type: Rolling
+      maxSurge: 50%
+      maxUnavailable: 0
+    template:
+      metadata:
+        labels:
+          name: ${COMPONENT_NAME}
+      spec:
+        containers:
+          # due to permission issue not using tools space; dev is pulling from dev
+          - image: >-
+              image-registry.openshift-image-registry.svc:5000/${OC_NAMESPACE}-${OC_ENV}/${APP_NAME}:${OC_ENV}
+            name: ${COMPONENT_NAME} 
+            imagePullPolicy: Always
+            ports:
+              - containerPort: 8080
+            env:
+            - name: SPLUNK_HTTP_URL
```

Comment:
dont think nginx will log to splunk.

Reply:
Depends:
(1) If there is corresponding setup at Gov end and (2) if we want to.
We will see/shall find that out.

ISB recommends using Splunk as a single pane of glass. Thanks!